### PR TITLE
[bug] Fix oneDALConfig.cmake.in for daal examples using a daal-only build

### PR DESCRIPTION
**Pull request by icfaust**
# Description
using a "make daal" oneDAL build with the daal examples will fail due to a missing onedal_parameters library. daal does not depend on this library, and so it erroneously will cause the cmake generation to fail. This change allows for daal-only builds to run examples.

Changes proposed in this pull request:
- Add check for ONEDAL_INTERFACE for onedal_parameters

### CODE REVIEW

File: cmake/templates/oneDALConfig.cmake.in [hunk 1]
```diff
@@ -121,7 +121,7 @@ elseif (ONEDAL_USE_DPCPP STREQUAL "yes" AND ONEDAL_INTERFACE STREQUAL "yes")
 endif()
 
 list(APPEND DAL_LIBS onedal_core${DAL_DEBUG_SUFFIX})
-if(ONEDAL_USE_PARAMETERS_LIBRARY)
+if(ONEDAL_USE_PARAMETERS_LIBRARY AND ONEDAL_INTERFACE STREQUAL "yes")
```
Comment (napetrov) [inline cmake/templates/oneDALConfig.cmake.in:124]: I think we can move  this entire section to elseif section above ?
elseif (ONEDAL_USE_DPCPP STREQUAL "no" AND ONEDAL_INTERFACE STREQUAL "yes")

Comment (icfaust) [reply cmake/templates/oneDALConfig.cmake.in:124]: I'll swap it and see if I can run the dpc examples on their own without onedal_parameters

Comment (napetrov) [reply cmake/templates/oneDALConfig.cmake.in:124]: the main idea is to have single if group rather than multiple through the file

**Review by Vika-F — APPROVED** [review]
The changes look good to me.
But there is a code freeze for oneDAL 2024.0.1 currently.
If those changes can be merged after the CF, please wait until the CF passes.

**Comment by icfaust** [comment]
Good question, the cleanest way I would imagine would be to split the build step for daal and oneDAL, then test for daal-specific stuff in between.

**Pull request merged**